drivers/docker/fingerprint.go

```diff
@@ -47,7 +67,9 @@ func (d *Driver) buildFingerprint() *drivers.Fingerprint {
 
 	env, err := client.Version()
 	if err != nil {
-		d.logger.Debug("could not connect to docker daemon", "endpoint", client.Endpoint(), "error", err)
+		if !d.fingerprinted() {
+			d.logger.Debug("could not connect to docker daemon", "endpoint", client.Endpoint(), "error", err)
```

Comment:
We maybe want to log this, when transitioning from a healthy to unhealthy state?

Reply:
Rather than implement such logic in every driver, could we do this in the client in a generic way based on the fingerprint response? The `HealthDescription` could contain the detailed error message when its state is failed

Reply:
@dantoml this is ready for another look. I changed drivers that were doing logging in their fingerprint method, not every driver needed changes.